File: app/src/main/java/com/lzy/testproject/other/reflect/Person.java
package com.lzy.testproject.other.reflect;

/**
 * Created by LiZhiyu on 2018/10/31.
 */
public class Person {
    public Person() {
        System.out.println("无参构造函数");
    }

    public Person(String name) {
        this.name = name;
        System.out.println("有参数构造函数" + name);
    }

    private String name;

    private void study(String name) {
        System.out.println("调用有参数方法" + name);
    }

    public static void method(){
        System.out.println("调用静态方法");
    }
}

File: app/src/main/java/com/lzy/testproject/kotlin/Person.kt
package com.lzy.testproject.kotlin

/**
 * Created by LiZhiyu on 2018/5/31.
 */
class Person(var name: String) {
    var age: Int = 2
    val sex: String
        get() = if (name == "呵呵") "男" else "女"


    constructor(name: String, age: Int) : this(name) {

    }
}